src/main/java/org/cadixdev/mercury/remapper/SimpleRemapperVisitor.java

```diff
@@ -93,6 +102,47 @@ private void remapField(SimpleName node, IVariableBinding binding) {
         updateIdentifier(node, mapping.getDeobfuscatedName());
     }
 
+    private void remapParameter(SimpleName node, IVariableBinding binding) {
+        IMethodBinding declaringMethod = binding.getDeclaringMethod();
+
+        if (declaringMethod == null) {
+           return;
+        }
+
+        int index = -1;
+
+        for (ASTNode n = node; n != null; n = n.getParent()) {
+            if (n instanceof MethodDeclaration) {
```

Comment:
This isn't safe - if a variable is used inside a lambda expression or anonymous class (either by being `final` or being effectively final, passed as a closure value) then the method you'll find first will be that method instead of the actual declaring method.

A better and more direct method is to simply resolve the declaring method binding like I do in paperweight: https://github.com/DemonWav/paperweight/blob/master/src/main/kotlin/tasks/RemapSources.kt#L208-L220

Reply:
Note the `context.getCompilationUnit().findDeclaringNode(method)` call is the relevant part.